src/data/signature-fetcher.ts

```diff
@@ -84,6 +89,40 @@ export class SignatureFetcher implements SignatureSource {
         id,
         error: (error as Error).message,
       });
+
+      return undefined;
+    }
+  }
+
+  async getTransactionSignature(id: string): Promise<string | undefined> {
+    try {
+      this.log.debug('Fetching transaction signature', { id });
+
+      const dataAttributes = await this.dataIndex.getDataAttributes(id);
```

Comment:
I think we should make a new `getTxAttributes` method for this rather than adding the signature `getDataAttributes`. I both don't want to pay the cost of all the queries in `getDataAttributes` here and don't want to add the overhead of fetching signatures in `getDataAttributes`.

Reply:
Make sense
[`240c0c4` (#187)](https://github.com/ar-io/ar-io-node/pull/187/commits/240c0c47d769d31de170db688e5dff81f17fc1a7)

Reply:
This adds the new method, but it looks like `signature` is still being returned from `getDataAttributes`. Unless we need that elsewhere, let's stop returning it from `getDataAttributes` (both the method and query results).

Reply:
[`229ada5` (#187)](https://github.com/ar-io/ar-io-node/pull/187/commits/229ada578e067cf229c52c15e6c75c571389d6d2)